#!/usr/bin/env python

PACKAGE = 'gripper_controller'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# 位置参数
gen.add("position_tolerance", double_t, 0, "Position tolerance", 0.001, 0.001, 0.1)

# 堵转检测参数
gen.add("stalled_velocity_threshold", double_t, 0, "Stalled velocity threshold", 0.01, 0.0001, 0.01)
gen.add("stalled_effort_threshold", double_t, 0, "Stalled effort threshold", 15.0, 0.1, 200.0)
gen.add("stall_timeout", double_t, 0, "Stall timeout (seconds)", 0.02, 0.05, 1.0)

gen.add("release_offset", double_t, 0, "Release position offset", 0.01, 0.001, 0.1)

exit(gen.generate(PACKAGE, PACKAGE, "GripperController"))
